index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/TransportAddress.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.jcr.index.elasticsearch;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import org.elasticsearch.common.transport.InetSocketTransportAddress;
+
+/**
+ *
+ * @author kulikov
+ */
+public class TransportAddress {
```

Comment:
This class is used for parsing addresses and ports of the ES cluster nodes (actually it is separated as dedicated class because it is easy to test its feature rather then implement/test as private method inside `EsIndexProvider` class). The default value is localhost:9300. No hardcoded  ports.